src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -326,18 +392,28 @@ private static void sendResponseMessages(RelayedHttpListenerContext context, Str
 		}
 	}
 	
-	private static String readStringFromStream(InputStream inputStream) throws IOException {
+	private static byte[] readBytesFromStream(InputStream inputStream) throws IOException {
 		if (inputStream == null) {
-			return emptyStr;
+			return new byte[0];
```

Comment:
> new byte[0]; [](start = 9, length = 13)

Food for thought: I'm curious if this has the result of converting a null inputStream, kind of like (String)null, into a "". Does this make it harder to test any variations we might be interested in?

Reply:
I did this to avoid potentially having null checks everywhere.

The only side effect I can think of is if we send and receive null vs. byte[0], but according to the code, sending null would just return and do nothing, so I don't think we need to worry about tests sending and receiving null vs. byte[0].

---
In reply to: [294387162](https://github.com/Azure/azure-relay-java/pull/47#discussion_r294387162) [](ancestors = 294387162)